### PR TITLE
NAS-125710 / 24.04 / Add account_attributes to auth.me output

### PR DESCRIPTION
**Pull request by anodos325**
This adds specific account-related flags to user credentials that webui can evaluate whether user is local, our special system administrator account, etc. This is implemented as a list so that we have capability to expand what is presented without having to modify API.

This commit also removes two indirect calls to `user.query` during `auth.authenticate`.

### CODE REVIEW

File: tests/api2/test_audit.py [hunk 1]
```diff
@@ -259,7 +259,7 @@ def test_token_login():
                         "credentials_data": {
                             "parent": {
                                 "credentials": "LOGIN_PASSWORD",
-                                "credentials_data": {"username": "root"},
+                                "credentials_data": {"username": "root", "account_attributes": ["LOCAL", "SYS_ADMIN"]},
```
Comment (themylogin) [inline tests/api2/test_audit.py:262]: Do we need this in the audit logs?

Comment (anodos325) [reply tests/api2/test_audit.py:262]: I'm kind of torn, there won't be many of these, but if you want me to I can remove from the credentials.dump() output.

**Review by themylogin — APPROVED** [review]

**Review by yocalebo — APPROVED** [review]

File: src/middlewared/middlewared/plugins/account_/2fa.py [hunk 2]
```diff
@@ -29,7 +29,9 @@ async def provisioning_uri(self, username):
         """
         user = await self.translate_username(username)
         user_twofactor_config = await self.middleware.call(
-            'auth.twofactor.get_user_config', user['id' if user['local'] else 'sid'], user['local'],
+            'auth.twofactor.get_user_config',
+            user['id' if user['local'] else 'sid'],
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/account_/2fa.py:33]: I don't like these types of comprehensions when we're checking the same variable twice. It's also kind of messy code. What do you think about doing something like:
```
user_id, local_user = 'id', 'DB_ID'
if not user['local']:
    user_id, local_user = 'sid', 'SID'

user_twofactor_config = await self.middleware.call(
    'auth.twofactor.get_user_config', user_id, local_user
)

Comment (anodos325) [reply src/middlewared/middlewared/plugins/account_/2fa.py:33]: I'm actually testing backing out the 2fa changes right now.

**Pull request merged**